llvm/include/llvm/Analysis/TargetTransformInfoImpl.h

```diff
@@ -32,6 +32,8 @@ class Function;
 /// Base class for use as a mix-in that aids implementing
 /// a TargetTransformInfo-compatible class.
 class TargetTransformInfoImplBase {
+  friend class TargetTransformInfo;
```

Comment:
I just wondered why this is needed?

Reply:
It's a leftover from a previous attempt at making an overloaded method to call the existing fixed-only method from the base implementation if there wasn't an override of the newer method. It didn't work out, so I've removed it. Thanks for spotting it.